rs-matter/src/data_model/cluster_window_covering.rs

```diff
@@ -0,0 +1,658 @@
+        Attribute::new(
+            AttributesDiscriminants::ConfigStatus as u16,
+            Access::RV,
+            Quality::N,
+        ),
+        Attribute::new(
+            AttributesDiscriminants::OperationalStatus as u16,
+            Access::RV,
+            Quality::P,
+        ),
+        Attribute::new(
+            AttributesDiscriminants::EndProductType as u16,
+            Access::RV,
+            Quality::F,
+        ),
+        Attribute::new(
+            AttributesDiscriminants::Mode as u16,
+            Access::RWVM,
+            Quality::N,
+        ),
+    ],
+    commands: &[
+        CommandsDiscriminants::UpOrOpen as _,
+        CommandsDiscriminants::DownOrClose as _,
+        CommandsDiscriminants::StopMotion as _,
+        CommandsDiscriminants::GoToLiftValue as _,
+        CommandsDiscriminants::GoToLiftPercentage as _,
+        CommandsDiscriminants::GoToTiltValue as _,
+        CommandsDiscriminants::GoToTiltPercentage as _,
+    ],
+};
+
+pub struct WindowCoveringCluster {
+    data_ver: Dataver,
+    type_attribute: Cell<TypeAttribute>,
+    config_status: Cell<ConfigStatus>,
+    operational_status: Cell<OperationalStatus>,
+    end_product_type: Cell<EndProductType>,
+    mode: Cell<Mode>,
+}
+
+impl WindowCoveringCluster {
+    pub fn new(rand: Rand) -> Self {
+        Self {
+            data_ver: Dataver::new(rand),
+            type_attribute: Cell::new(TypeAttribute::default()),
+            config_status: Cell::new(ConfigStatus::default()),
+            operational_status: Cell::new(OperationalStatus::default()),
+            end_product_type: Cell::new(EndProductType::default()),
+            mode: Cell::new(Mode::default()),
+        }
+    }
+
+    pub fn set_type(&self, device_type: TypeAttribute) {
+        if self.type_attribute.get() != device_type {
+            self.type_attribute.set(device_type);
+            self.data_ver.changed();
+        }
+    }
+
+    pub fn read(&self, attr: &AttrDetails, encoder: AttrDataEncoder) -> Result<(), Error> {
+        if let Some(writer) = encoder.with_dataver(self.data_ver.get())? {
+            if attr.is_system() {
+                CLUSTER.read(attr.attr_id, writer)
+            } else {
+                match attr.attr_id.try_into()? {
+                    Attributes::Type(codec) => codec.encode(writer, self.type_attribute.get()),
+                    Attributes::ConfigStatus(codec) => {
+                        codec.encode(writer, self.config_status.get())
+                    }
+                    Attributes::OperationalStatus(codec) => {
+                        codec.encode(writer, self.operational_status.get())
+                    }
+                    Attributes::EndProductType(codec) => {
+                        codec.encode(writer, self.end_product_type.get())
+                    }
+                    Attributes::Mode(codec) => codec.encode(writer, self.mode.get()),
+                    _ => todo!("read {:?}", attr.attr_id),
```

Comment:
Would that crash on any other attribute and is this the intended behavior?

Reply:
If the attribute is currently not supported I'd expect the function to return such an error